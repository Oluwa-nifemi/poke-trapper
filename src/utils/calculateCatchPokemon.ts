const calculateCatchPokemon = (): boolean => Math.random() > 0.5

export default calculateCatchPokemon