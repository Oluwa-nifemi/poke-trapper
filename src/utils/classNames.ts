/* This function allows you to merge several classes together.
 * For conditional classes the first item in the array should be the condition (boolean)
 */
export const classNames = (...args: (string | [boolean, string])[]) => {
    return args.filter(className => {
        if (Array.isArray(className)) {
            return className[0];
        }

        return className;
    }).map((className) => {
        if (Array.isArray(className)) {
            return className[1];
        }

        return className;
    }).join(' ')
};