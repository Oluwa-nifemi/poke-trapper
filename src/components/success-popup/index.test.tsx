import {render, screen} from "tests/utils";
import SuccessPopup from "./index";
import userEvent from "@testing-library/user-event";
import {waitFor} from "@testing-library/react";

const mockCloseFunction = jest.fn();

describe("Success Popup", () => {
    it("Renders and allows submit", async () => {
        const user = userEvent.setup()

        const {container} = render(
            <SuccessPopup
                pokemonName="bulbasaur"
                handleClose={mockCloseFunction}
                imageUrl="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
            />
        )

        expect(container).toMatchSnapshot();

        await user.click(screen.getByText("Close"))

        expect(mockCloseFunction).toHaveBeenCalledTimes(1);

        expect(screen.getByText("Save")).toBeDisabled()

        const testNickname = "Test Nickname";

        await user.type(screen.getByLabelText("Enter Nickname"), testNickname)

        expect(screen.getByText("Save")).not.toBeDisabled()

        await user.click(screen.getByText("Save"))

        await waitFor(() => screen.findByTitle(`${testNickname} was saved!`))
        expect(container).toMatchSnapshot()
    });
});