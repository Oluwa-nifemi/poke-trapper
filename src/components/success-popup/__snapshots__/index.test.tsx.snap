// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success Popup Renders and allows submit 1`] = `
<div>
  <div
    class="popup isVisible successPopup"
    role="alert"
  >
    <p
      title="bulbasaur was caught!"
    >
      <span
        class="letter"
        style="animation-delay: 0s;"
      >
        b
      </span>
      <span
        class="letter"
        style="animation-delay: 0.06s;"
      >
        u
      </span>
      <span
        class="letter"
        style="animation-delay: 0.12s;"
      >
        l
      </span>
      <span
        class="letter"
        style="animation-delay: 0.18s;"
      >
        b
      </span>
      <span
        class="letter"
        style="animation-delay: 0.24s;"
      >
        a
      </span>
      <span
        class="letter"
        style="animation-delay: 0.3s;"
      >
        s
      </span>
      <span
        class="letter"
        style="animation-delay: 0.36s;"
      >
        a
      </span>
      <span
        class="letter"
        style="animation-delay: 0.42s;"
      >
        u
      </span>
      <span
        class="letter"
        style="animation-delay: 0.48s;"
      >
        r
      </span>
      <span
        class="letter"
        style="animation-delay: 0.54s;"
      >
         
      </span>
      <span
        class="letter"
        style="animation-delay: 0.6s;"
      >
        w
      </span>
      <span
        class="letter"
        style="animation-delay: 0.6599999999999999s;"
      >
        a
      </span>
      <span
        class="letter"
        style="animation-delay: 0.72s;"
      >
        s
      </span>
      <span
        class="letter"
        style="animation-delay: 0.78s;"
      >
         
      </span>
      <span
        class="letter"
        style="animation-delay: 0.84s;"
      >
        c
      </span>
      <span
        class="letter"
        style="animation-delay: 0.8999999999999999s;"
      >
        a
      </span>
      <span
        class="letter"
        style="animation-delay: 0.96s;"
      >
        u
      </span>
      <span
        class="letter"
        style="animation-delay: 1.02s;"
      >
        g
      </span>
      <span
        class="letter"
        style="animation-delay: 1.08s;"
      >
        h
      </span>
      <span
        class="letter"
        style="animation-delay: 1.14s;"
      >
        t
      </span>
      <span
        class="letter"
        style="animation-delay: 1.2s;"
      >
        !
      </span>
    </p>
    <form
      class="successPopupForm"
      style="animation-delay: 1.36s;"
    >
      <label
        for="nickname"
      >
        <span
          class="successPopupInputLabel"
        >
          Enter Nickname
        </span>
        <input
          class="successPopupInput"
          id="nickname"
          type="text"
          value=""
        />
      </label>
      <button
        class="button"
        disabled=""
        type="submit"
      >
        Save
      </button>
      <button
        class="button"
        type="button"
      >
        Close
      </button>
    </form>
  </div>
</div>
`;

exports[`Success Popup Renders and allows submit 2`] = `
<div>
  <div
    class="popup isVisible successPopup"
    role="alert"
  >
    <p
      title="Test Nickname was saved!"
    >
      <span
        class="letter"
        style="animation-delay: 0s;"
      >
        T
      </span>
      <span
        class="letter"
        style="animation-delay: 0.06s;"
      >
        e
      </span>
      <span
        class="letter"
        style="animation-delay: 0.12s;"
      >
        s
      </span>
      <span
        class="letter"
        style="animation-delay: 0.18s;"
      >
        t
      </span>
      <span
        class="letter"
        style="animation-delay: 0.24s;"
      >
         
      </span>
      <span
        class="letter"
        style="animation-delay: 0.3s;"
      >
        N
      </span>
      <span
        class="letter"
        style="animation-delay: 0.36s;"
      >
        i
      </span>
      <span
        class="letter"
        style="animation-delay: 0.42s;"
      >
        c
      </span>
      <span
        class="letter"
        style="animation-delay: 0.48s;"
      >
        k
      </span>
      <span
        class="letter"
        style="animation-delay: 0.54s;"
      >
        n
      </span>
      <span
        class="letter"
        style="animation-delay: 0.6s;"
      >
        a
      </span>
      <span
        class="letter"
        style="animation-delay: 0.6599999999999999s;"
      >
        m
      </span>
      <span
        class="letter"
        style="animation-delay: 0.72s;"
      >
        e
      </span>
      <span
        class="letter"
        style="animation-delay: 0.78s;"
      >
         
      </span>
      <span
        class="letter"
        style="animation-delay: 0.84s;"
      >
        w
      </span>
      <span
        class="letter"
        style="animation-delay: 0.8999999999999999s;"
      >
        a
      </span>
      <span
        class="letter"
        style="animation-delay: 0.96s;"
      >
        s
      </span>
      <span
        class="letter"
        style="animation-delay: 1.02s;"
      >
         
      </span>
      <span
        class="letter"
        style="animation-delay: 1.08s;"
      >
        s
      </span>
      <span
        class="letter"
        style="animation-delay: 1.14s;"
      >
        a
      </span>
      <span
        class="letter"
        style="animation-delay: 1.2s;"
      >
        v
      </span>
      <span
        class="letter"
        style="animation-delay: 1.26s;"
      >
        e
      </span>
      <span
        class="letter"
        style="animation-delay: 1.3199999999999998s;"
      >
        d
      </span>
      <span
        class="letter"
        style="animation-delay: 1.38s;"
      >
        !
      </span>
    </p>
    <button
      class="button"
    >
      Close
    </button>
  </div>
</div>
`;
